### ISSUE TITLE
HDR Monitors - Incorrect Colors Displayed

### ISSUE THREAD

**Issue by Ziflin** [darktable-org/darktable#15967]
### Describe the bug

I'm using a pair of LG HDR monitors in Windows 11 with HDR enabled in the System / Display settings ("Use HDR" is enabled. "Auto HDR" is disabled. They were calibrated using the Windows HDR Calibration tool and seem to be working properly.)

When HDR is enabled in Windows, Darktable's Lighttable and Darkroom tabs display photos with a significant blue-green tint (lack of red). This is very apparent when creating a black and white image using the Monochrome, Color Balance RGB, etc. modules. There is about 10% less red displayed than blue and green (blue and green values match), though if I use the "Color Picker" tool in Darktable, it does show equal RGB values.

If I export an image and bring it into the Windows "Photo" app, Affinity Photo, MS Paint, etc. it is displayed correctly.

If I _disable_ HDR monitor support in Windows and restart Darktable, Darktable will then properly display the images.

Notes:
* I did notice that if I move the Monochrome module *above* the Output color profile module, that it will properly display gray-scale colors even with HDR enabled, but this obviously doesn't fix color photos and thumbnails are sill incorrectly displayed.
* If I export a Monochrome TIFF photo and use the *old* "Windows Photo Viewer" app, it too displays the image with the blue-green tint. This is the only app that appears to do this, so perhaps it has to do with how the window is being initialized? (sorry, I haven't looked into how the Windows API supports HDR yet).

The two attached screenshots show what I'm seeing in Darktable 4.6.0:

With "HDR" enabled in Windows Display Settings:
![Darktable-HDR-Monitor-On](https://github.com/darktable-org/darktable/assets/5809860/5eb45ebb-06ad-445e-bac2-7ff21ab43cfa)

With "HDR" disabled in Windows Display Settings:
![Darktable-HDR-Monitor-Off](https://github.com/darktable-org/darktable/assets/5809860/623aea96-455a-45c0-8f8d-af41b1995967)

### Steps to reproduce

1. Using an HDR capable monitor, enable "Use HDR" in Windows System/Display settings.
2. Start Darktable and enable the Monochrome module and note the blue-green tint.

### Expected behavior

Darktable should support HDR monitors with similar results to SDR ones.

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.6.0

### What OS are you using?

Windows

### What is the version of your OS?

Windows 11 Pro

### Describe your system?

Windows 11 Pro (23H2), Dual LG HDR Monitors (27GP83B-B), AMD 7800 XT GPU

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

AMD 7800 XT (16GB v23.11.1)

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

I noticed this in Darktable v4.4.2 (the first version that I have used).
The steps are reproducible with a clean edit and only enabling the Monochrome module to make it obvious.
Yes, the issue was still preset with a clean Darktable config.

**Comment by jenshannoschwalm** [comment]
Whatever the term "HDR monitor" exactly means, in dt you have to set the display profile :-) See online manual https://darktable-org.github.io/dtdocs/en/special-topics/color-management/display-profile/

**Comment by Ziflin** [comment]
@jenshannoschwalm Changing the "Preferences / Miscellaneous / Method to use for getting the display profile" setting to either of the other two options has no effect on Windows. The documentation does not mention Windows HDR settings anywhere that I can see.

**Comment by Ziflin** [comment]
Also, I'm not sure what the confusion over HDR Monitor means, but in Windows, having a monitor with HDR support allows you to enable the HDR setting which enables HDR video streams on YouTube, in-game HDR support, etc.  There's a bit more info Windows related info [here](https://www.dpreview.com/forums/thread/4728056), but what seems to work for Darktable is to right-click and edit the `darktable.exe` properties. Under the "Compatibility" tab, there is an option to "Use legacy display ICC color management".

Enabling this setting fixed the issues I had with Darktable when Windows HDR setting is enabled as I'm assuming Darktable hasn't updated its color management to whatever Windows now wants. I guess you can close this issue now. Oh wait, I guess you already did even though your answer was not useful.

**Comment by jenshannoschwalm** [comment]
Darktable always assumes a defined monitor colorspace so same output on all platforms.

**Comment by wpferguson** [comment]
> Oh wait, I guess you already did even though your answer was not useful.

The only way we could fix this is if we had a windows developer with an HDR monitor setup similar to yours.  AFAIK we don't have any windows developers that use windows as their primary system.  All the devs that maintain the windows parts either use spare hardware, dual boot, or do it in a VM.

>  I'm assuming Darktable hasn't updated its color management to whatever Windows now want

I'm not sure that any of the windows maintainers even know that windows has an HDR setting (I didn't).

**Comment by ptilopteri** [comment]
perhaps a "windows" <user> will happily donate an HDR windows monitor to the project.  (tongue in cheek).

**Comment by jenshannoschwalm** [comment]
I couldn't even find a spec...

**Comment by Ziflin** [comment]
Thanks @wpferguson. Your response was what I was trying to determine. @jenshannoschwalm initial reply just made it sound like I had missed an obvious setting to fix the issue (which I did, just not a Darktable one). From what I can tell, Windows has had quite a few problems with HDR support (including their own screen snipping tool until their update a week or so ago and broken API calls).

And apparently this issue is another one... From item 8 in this [article](https://devblogs.microsoft.com/directx/the-windows-hdr-calibration-app-is-here/) regarding the release of the "Windows HDR Calibration" app (that I had used for proper brightness calibration):

> 8. I just calibrated my display and now the images in Adobe Photoshop and Lightroom are appearing **green**, is this normal?

With the solution being to check the ICC color management compatibility option I mentioned above *or* **delete** the HDR color profile that their "Calibration" app creates.... **sigh**.

Maybe this will help anyone else that runs into it until MS can fix their calibration app. (I think I'll just disable HDR for now...)
